pkg/inject/inject.go

```diff
@@ -599,24 +615,41 @@ func (conf *ResourceConfig) injectPodSpec(patch *Patch) {
 }
 
 func (conf *ResourceConfig) injectProxyInit(patch *Patch, saVolumeMount *corev1.VolumeMount) {
-	nonRoot := false
-	runAsUser := int64(0)
+	capabilities := &corev1.Capabilities{}
+	if conf.pod.spec.Containers != nil && len(conf.pod.spec.Containers) > 0 {
+		if sc := conf.pod.spec.Containers[0].SecurityContext; sc != nil && sc.Capabilities != nil {
+			capabilities = sc.Capabilities
+		}
+	}
+	if capabilities.Add == nil {
+		capabilities.Add = []corev1.Capability{}
+	}
+	capabilities.Add = append(capabilities.Add, corev1.Capability("NET_ADMIN"))
```

Comment:
Unfortunately this seems to break proxy-init. Adding NET_RAW to the primary container's capabilities works around it. Should we add NET_RAW here? #2959

Reply:
Can confirm that `NET_RAW` is needed when either the PSP or primary container security context has a `Drop: ALL` capability rule.

For posterity, the reason why we have to copy the capabilities of the primary container into the proxy-init and proxy sidecars is because by the time the proxy injector receives the admission request, the primary container would have been mutated by the PSP admission controller to include any default `requiredDropCapabilities` rules. The PSP validation which happens after expects the rules to be present in all the containers' security context. AFACIT, the validation is performing a exact string values match.

All these will become moot in k8s 1.15 per https://github.com/linkerd/linkerd2/issues/2767#issuecomment-497538600